foxpuppet/windows/browser/notifications/addons.py

```diff
@@ -59,14 +59,57 @@ def close(self):
 class AddOnInstallRestart(BaseNotification):
     """Add-on install restart notification."""
 
+    def restart_now(self):
+        """Restart Firefox immediately to complete the add-on installation."""
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            self.find_primary_button().click()
+
+    def restart_later(self):
+        """Defer Firefox restart for later."""
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            self.find_secondary_button().click()
+
 
 class AddOnInstallFailed(BaseNotification):
     """Add-on install failed notification."""
 
+    @property
+    def error_message(self):
+        """Provide access to the error message.
+
+        Returns:
+            str: The error message explaining why the installation failed.
+        """
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            return self.find_description().text
+
+    def close(self):
+        """Close the failed installation notification."""
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            self.find_primary_button().click()
+
 
 class AddOnProgress(BaseNotification):
     """Add-on progress notification."""
 
+    @property
+    def is_downloading(self):
+        """Check if the add-on is currently downloading.
+
+        Returns:
+            bool: True if the download and verification is in progress.
+        """
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            return "Downloading and verifying add-on…" in self.find_description().text
```

Comment:
So what's the idea with this. I can see some use cases but I would like to know you're thinking. How would the API look for a user who wants to use this?

Reply:
I added this as property here because I think this is a good identifier of the Firefox Progress notification. The idea is for the property to return a boolean. A simple true or false could be used to easily determine if the Progress notification is active or not. In the context of the test for notification, if `progress_notification` returns the `AddOnProgress` class, then the API will be `progress_notification.is_downloading`.